diff -ruNb a/urlgrabber/grabber.py b/urlgrabber/grabber.py
--- a/urlgrabber/grabber.py	2009-09-25 17:19:50.000000000 +0100
+++ b/urlgrabber/grabber.py	2013-05-26 13:03:07.136563659 +0100
@@ -1157,8 +1157,10 @@
             if opts.ssl_ca_cert: # this may do ZERO with nss  according to curl docs
                 self.curl_obj.setopt(pycurl.CAPATH, opts.ssl_ca_cert)
                 self.curl_obj.setopt(pycurl.CAINFO, opts.ssl_ca_cert)
-            self.curl_obj.setopt(pycurl.SSL_VERIFYPEER, opts.ssl_verify_peer)
-            self.curl_obj.setopt(pycurl.SSL_VERIFYHOST, opts.ssl_verify_host)
+            if opts.ssl_verify_host:
+                self.curl_obj.setopt(pycurl.SSL_VERIFYHOST, 2)
+            else:
+                self.curl_obj.setopt(pycurl.SSL_VERIFYHOST, 0)
             if opts.ssl_key:
                 self.curl_obj.setopt(pycurl.SSLKEY, opts.ssl_key)
             if opts.ssl_key_type:
